src/mono/System.Private.CoreLib/src/System/Reflection/AssemblyName.Mono.cs

```diff
@@ -19,6 +19,9 @@ public AssemblyName(string assemblyName)
             if (assemblyName.Length == 0 || assemblyName[0] == '\0')
                 throw new ArgumentException(SR.Format_StringZeroLength);
 
+            if (assemblyName.Contains('\0'))
```

Comment:
This check is just to match CoreClr behavior for now and pass tests. Once this API is unified, this entire constructor will be gone.